src/main/java/ai/aliz/jarvis/context/ContextLoader.java

```diff
@@ -0,0 +1,55 @@
+package ai.aliz.jarvis.context;
+
+import lombok.Getter;
+import lombok.SneakyThrows;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.Files;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class ContextLoader {
+    
+    private final ObjectMapper objectMapper;
+    
+    public ContextLoader(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+    
+    @Getter
+    private Set<Context> contexts;
+    
+    @SneakyThrows
+    public void parseContext(String contextPath) {
```

Comment:
can you refactor this class in a way that the parseContext is automatically executed in the constructor and the contexts are final, making this component immutable

Reply:
Done.